presto-main/src/test/java/com/facebook/presto/type/BenchmarkBigIntOperators.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.LongBinaryOperator;
+import java.util.function.LongUnaryOperator;
+
+@State(Scope.Thread)
+@Fork(2)
+@Warmup(iterations = 100, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 100, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.Throughput)
+public class BenchmarkBigIntOperators
+{
+    private long leftOperand0;
+    private long leftOperand1;
+    private long leftOperand2;
+    private long leftOperand3;
+    private long leftOperand4;
+
+    private long rightOperand0;
+    private long rightOperand1;
+    private long rightOperand2;
+    private long rightOperand3;
+    private long rightOperand4;
+
+    @Setup
+    public void setup()
+    {
+        leftOperand0 = 1;
+        leftOperand1 = 20;
+        leftOperand2 = 33;
+        leftOperand3 = 407;
+        leftOperand4 = 7890;
+
+        rightOperand0 = 123456;
+        rightOperand1 = 9003;
+        rightOperand2 = 809;
+        rightOperand3 = 67;
+        rightOperand4 = 5;
+    }
+
+    @Benchmark
+    public Object overflowChecksAdd()
+    {
+        return execute(BigintOperators::add);
```

Comment:
Yeah, we changed them because that's how arithmetic operations are expected to work (and also protect against a class of correctness problems).

I don't think we should implement the ***_with_overflow functions. It's unlikely anyone would use them, and it's not clear how much the effective savings would be (due to other factors dominating the execution of queries)